Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +144,28 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null || array.Length == 0)
+            {
+                return array;
+            }
+
+            int[] tempArr = new int[array.Length + 1];
+            int oldArrayIndex = 0;
+            for (int i = 0; i < tempArr.Length; i++)
+            {
+                if (i == index)
+                {
+                    tempArr[i] = number;
+                    i++;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not update the loop counter 'i' within the loop body.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5075368)